sycl/test/ordered_queue/ordered_prop_dmemll.cpp

```diff
@@ -0,0 +1,116 @@
+// RUN: %clangxx -fsycl %s -o %t1.out -lOpenCL
+// RUN: %CPU_RUN_PLACEHOLDER %t1.out
+// RUN: %GPU_RUN_PLACEHOLDER %t1.out
+
+//==----------- ordered_dmemll.cpp - Device Memory Linked List test --------==//
+// It uses an ordered queue where explicit waiting is not necessary between
+// kernels
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+
+using namespace cl::sycl;
+
+constexpr int numNodes = 4;
+
+bool getQueueOrder(cl_command_queue cq) {
```

Comment:
Should we make this easier, so users can query it without resorting to the property list or OpenCL?

I'd be tempted to replace/augment `sycl::detail::getQueueOrder` with a member function of queue, something like `bool is_ordered()`.

Reply:
I thought about that.  It would effectively be an alias for the property query.  Not sure it's necessary as it's super easy to write yourself.

Reply:
+1
and leave only `in_order` property. Right?